bootstrap.py

```diff
@@ -426,19 +426,28 @@ def fully_update_the_box():
 
 # curl https://satellite.example.com:9090/ssh/pubkey >> ~/.ssh/authorized_keys
 # sort -u ~/.ssh/authorized_keys
-def install_foreman_ssh_key():
+def install_foreman_ssh_key(remote_url):
     """
     Download and install the Satellite's SSH public key into the foreman user's
-    authorized keys file, so that remote execution becomes possible.
+    authorized keys file location, so that remote execution becomes possible.
+    By default is ~/.ssh/authorized_keys
     """
     userpw = pwd.getpwnam(options.remote_exec_user)
-    foreman_ssh_dir = os.sep.join([userpw.pw_dir, '.ssh'])
-    foreman_ssh_authfile = os.sep.join([foreman_ssh_dir, 'authorized_keys'])
+    if options.remote_exec_authpath:
+        if os.path.isdir(options.remote_exec_authpath):
+            foreman_ssh_dir = options.remote_exec_authpath
+            foreman_ssh_authfile = options.remote_exec_authpath + "/authorized_keys"
```

Comment:
There is no requirement for the file to be called `authorized_keys`. In fact, some of my boxes have `AuthorizedKeysFile /etc/ssh/authorized_keys/%u` (with `%u` being the username). So I think we just should take the full path at the command line and not alter it at all.

Reply:
OK